dandi/tests/test_models.py

```diff
@@ -154,3 +166,265 @@ def test_autogenerated_titles():
+                    1,
+                    {"description": "testing", "descriptionType": "Abstract"},
+                ),
+                "publisher": (None, "DANDI Archive"),
+                "rightsList": (
+                    1,
+                    {"rightsIdentifierScheme": "SPDX", "rightsIdentifier": "CC_BY_40"},
+                ),
+                "types": (
+                    None,
+                    {"resourceType": "NWB", "resourceTypeGeneral": "Dataset"},
+                ),
+            },
+        ),
+        # additional contributor with dandi:Author
+        (
+            {
+                "contributor": [
+                    {
+                        "name": "A_last, A_first",
+                        "roleName": [RoleType("dandi:ContactPerson")],
+                    },
+                    {"name": "B_last, B_first", "roleName": [RoleType("dandi:Author")]},
+                ],
+            },
+            {
+                "creators": (1, {"name": "B_last, B_first"}),
+                "contributors": (
+                    1,
+                    {"name": "A_last, A_first", "contributorType": "ContactPerson"},
+                ),
+            },
+        ),
+        # additional contributor with dandi:Sponsor, fundingReferences should be created
+        (
+            {
+                "contributor": [
+                    {
+                        "name": "A_last, A_first",
+                        "roleName": [RoleType("dandi:ContactPerson")],
+                    },
+                    {
+                        "name": "B_last, B_first",
+                        "roleName": [RoleType("dandi:Sponsor")],
+                    },
+                ],
+            },
+            {
+                "creators": (1, {"name": "A_last, A_first"}),
+                "fundingReferences": (1, {"funderName": "B_last, B_first"}),
+            },
+        ),
+        # additional contributor with 2 roles: Author and Software (doesn't exist in datacite)
+        # the person should be in creators and contributors (with contributorType Other)
+        (
+            {
+                "contributor": [
+                    {
+                        "name": "A_last, A_first",
+                        "roleName": [
+                            RoleType("dandi:Author"),
+                            RoleType("dandi:Software"),
+                        ],
+                    },
+                    {
+                        "name": "B_last, B_first",
+                        "roleName": [RoleType("dandi:ContactPerson")],
+                    },
+                ],
+            },
+            {
+                "creators": (1, {"name": "A_last, A_first"}),
+                "contributors": (
+                    2,
+                    {"name": "A_last, A_first", "contributorType": "Other"},
+                ),
+            },
+        ),
+    ],
+)
+def test_dantimeta_datacite(schema, additional_meta, datacite_checks):
```

Comment:
This test should be skipped if `DATACITE_DEV_PASSWORD` is not set so that the test suite can be run without needing that secret.